### PR TITLE
Add W3C Error Response

### PR DESCRIPTION
**Pull request by dpgraham**
* Refactored `getResponseForW3CError` to return proper response 
  * Changed method so that it returns `{value: {stacktrace: '...', message: '...', error: '...'}}` (a la https://github.com/jlipps/simple-wd-spec#error-handling)
    * `message` is the same as the JSONWP error message
    * `error` is a static message indicating the type of error
    * `stacktrace` is the callstack provided in the error
  * Added an attribute called `error` to all of the MJSONWPErrors in `errors.js` which contains the static description of the W3C error
  * Also made `getResponseForW3CError` handle BadParametersError
    * `BadParametersError` isn't an `MJSONWPError` so it needs to be handled differently
* Refactored `getResponseForW3CError` and `getResponseForJsonwpError` into errors.js file

(connected to https://github.com/appium/appium/issues/10042)

### CODE REVIEW

File: lib/mjsonwp/errors.js [hunk 1]
```diff
@@ -1,17 +1,24 @@
 import ES6Error from 'es6-error';
 import _ from 'lodash';
-import { util } from 'appium-support';
+import { util, logger } from 'appium-support';
 import HTTPStatusCodes from 'http-status-codes';
 
+const log = logger.getLogger('MJSONWP');
+
 // base error class for all of our errors
 class MJSONWPError extends ES6Error {
```
Comment (mykola-mokhnach) [inline lib/mjsonwp/errors.js]: at some point of time we'll need to properly rename all this stuff

File: lib/mjsonwp/errors.js [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
-  constructor (msg, jsonwpCode, w3cStatus = 400) {
+  constructor (msg, jsonwpCode, w3cStatus, error) {
     super(msg);
     this.jsonwpCode = jsonwpCode;
+    this.error = error || 'An unknown error has occurred';
     if (this.jsonwpCode === null) {
       this.jsonwpCode = 13;
     }
-    this.w3cStatus = w3cStatus;
+    this.w3cStatus = w3cStatus || 400;
```
Comment (mykola-mokhnach) [inline lib/mjsonwp/errors.js]: can we have constants for these magic numbers?

File: lib/mjsonwp/errors.js [hunk 3]
```diff
@@ -1,17 +1,24 @@
...
+  }
+
+  static get error () {
```
Comment (mykola-mokhnach) [inline lib/mjsonwp/errors.js]: what is the purpose of this method?

File: lib/mjsonwp/errors.js (hunk 1 above)
Comment (dpgraham) [reply lib/mjsonwp/errors.js]: Yeah I should make an issue for that. Classes and files will need to be renamed.

File: lib/mjsonwp/errors.js [hunk 4]
```diff
@@ -299,9 +398,12 @@ class InvalidSelectorError extends MJSONWPError {
   static code () {
     return 32;
   }
+  static error () {
```
Comment (mykola-mokhnach) [inline lib/mjsonwp/errors.js]: I assume there should be at least one line space between class methods

File: lib/mjsonwp/errors.js (hunk 3 above)
Comment (dpgraham) [reply lib/mjsonwp/errors.js]: Does nothing. Will take it out.

File: lib/mjsonwp/errors.js (hunk 1 above)
Comment (dpgraham) [reply lib/mjsonwp/errors.js]: https://github.com/appium/appium/issues/10041

**Comment by dpgraham** [comment]
Published to 2.20.2

**Pull request merged**